doc/devel/coding_guide.rst

```diff
@@ -13,9 +13,89 @@
 Pull request guidelines
 ***********************
 
-Pull requests (PRs) are the mechanism for contributing to Matplotlibs code and
+Pull requests (PRs) are the mechanism for contributing to Matplotlib's code and
 documentation.
 
+It is recommended to check that your contribution complies with the following
+rules before submitting a pull request:
+
+* If your pull request addresses an issue, please use the title to describe the
+  issue and mention the issue number in the pull request description to ensure
+  that a link is created to the original issue.
+
+* All public methods should have informative docstrings with sample usage when
+  appropriate. Use the `numpy docstring standard
+  <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
```

Comment:
Better link https://matplotlib.org/stable/devel/documenting_mpl.html#writing-docstrings

and move this one bullet point down, so that we have

- formatting
- docstring standard
- docstring example section